lua/telescope/make_entry.lua

```diff
@@ -57,6 +57,40 @@ do
   end
 end
 
+do
+  local lookup_keys = {
+    ordinal = 1,
+    value = 2,
+    display = 1,
+    key = 1,
+  }
+
+  local mt_tree_entry = {
+    __index = function(t, k)
+      return rawget(t, rawget(lookup_keys, k))
+    end
+  }
+
+  function make_entry.gen_from_tree()
+    return function(key, value)
+      if type(key) == "number" then
+        -- it is a leaf
+        return setmetatable({
+          value,
+          -- this means it is leaf
+          nil,
+        }, mt_tree_entry)
+      else
+        -- it is a node
+        return setmetatable({
+          key,
+          value,
+        }, mt_tree_entry)
+      end
+    end
+  end
+end
```

Comment:
Also can be realized easier without a metatable.

Reply:
It may be a bit confusing using timetables, I agree. I was trying to keep the same style as the other entry maker functions use them too.

Reply:
These metatables are only used for builtins like `find_files`, `live_grep`, buitlins which eventually deal with 10000 and more entries and having display and ordinal the same for each entry requires a lot more ram usage, so it makes sense to have it with those `lookup_keys`. But you will end up with lets say 50 entrys max and so a simpler version is good enough in my opinion. You can find these simples once after the metatables. On `master` they start at line: 216. Or they can be found at the builtin picker definition itself, if it is even simpler (example: `telescope.builtin.internal.keymaps` starting line 553 on master)